### PR TITLE
fix: events not deleted on cancelling maintenance schedule

### PR DESCRIPTION
**Pull request by scmmishra**
Backport of https://github.com/frappe/erpnext/pull/22954

---

When cancelling a maintenance schedule, associated events would not be deleted. This was because they were not created properly in the first place. 

This PR fixes that

Depends On: https://github.com/frappe/frappe/pull/11221 [Merged]

### CODE REVIEW

File: erpnext/maintenance/doctype/maintenance_schedule/test_maintenance_schedule.py [hunk 1]
```diff
@@ -2,11 +2,47 @@
...
+from frappe.utils.data import get_datetime, add_days
 
 import frappe
 import unittest
 
 # test_records = frappe.get_test_records('Maintenance Schedule')
 
 class TestMaintenanceSchedule(unittest.TestCase):
-	pass
+	def events_should_be_created_and_deleted(self):
```
Comment (marination) [inline erpnext/maintenance/doctype/maintenance_schedule/test_maintenance_schedule.py]:
```suggestion
	def test_events_should_be_created_and_deleted(self):
```

**Review by marination — APPROVED** [review]

**Comment by Olawale1** [comment]
Hi @scmmishra 

Thanks for this patch. In our instance, the Events are not even getting created at all after submitting the Maintenance Schedule. Does this address the issue as well?

ERPNext: v12.10.1 (version-12)
Frappe Framework: v12.8.4 (version-12)

Kind regards,

**Comment by Olawale1** [comment]
So, it turns out the Event actually gets created but it's totally invisible to the Creator of the Maintenance Schedule and System Manager! Only the 'Sales Person' designated in the schedule can view the Events... I think this is a rather poor design and should probably be replaced with something similar to the Asset Maintenance implementation

Thanks

**Comment by Olawale1** [comment]
Created a Github Issue: https://github.com/frappe/erpnext/issues/23682

**Pull request merged**